modules/apps/blogs/blogs-web/src/main/resources/META-INF/resources/blogs_admin/view_images.jsp

```diff
@@ -110,56 +110,14 @@ String displayStyle = blogImagesManagementToolbarDisplayContext.getDisplayStyle(
 	</aui:form>
 </clay:container-fluid>
 
-<aui:script>
-	var deleteImages = function () {
-		if (
-			confirm(
-				'<liferay-ui:message key="are-you-sure-you-want-to-delete-the-selected-images" />'
-			)
-		) {
-			var form = document.getElementById('<portlet:namespace />fm');
-
-			if (form) {
-				var cmd = form.querySelector(
-					'#<portlet:namespace /><%= Constants.CMD %>'
-				);
-
-				if (cmd) {
-					cmd.setAttribute('value', '<%= Constants.DELETE %>');
-				}
-
-				var deleteFileEntryIds = form.querySelector(
-					'#<portlet:namespace />deleteFileEntryIds'
-				);
-
-				if (deleteFileEntryIds) {
-					deleteFileEntryIds.setAttribute(
-						'value',
-						Liferay.Util.listCheckedExcept(
-							form,
-							'<portlet:namespace />allRowIds'
-						)
-					);
-				}
-
-				submitForm(form);
-			}
-		}
-	};
-
-	var ACTIONS = {
-		deleteImages: deleteImages,
-	};
-
-	Liferay.componentReady('blogImagesManagementToolbar').then(function (
-		managementToolbar
-	) {
-		managementToolbar.on('actionItemClicked', function (event) {
-			var itemData = event.data.item.data;
-
-			if (itemData && itemData.action && ACTIONS[itemData.action]) {
-				ACTIONS[itemData.action]();
-			}
-		});
-	});
-</aui:script>
\ No newline at end of file
+<liferay-frontend:component
```

Comment:
I think the current pattern for this is to create a `defaultEventHandler` and pass it to the toolbar. This should change soon enough as we migrate that tag to React, we'll see...

Reply:
mmmmmm, good! I saw some examples of components extending the DefaultEventHandler but I didn't saw any documentation why people where doing this.